### ISSUE TITLE
[4.2.6] Missing parameter help

### ISSUE THREAD

**Issue by PhocaCz** [joomla/joomla-cms#39545]
### Steps to reproduce the issue

This issue is related to these issues:

https://github.com/joomla/joomla-cms/issues/39544
https://github.com/joomla/joomla-cms/discussions/35594
https://github.com/joomla/joomla-cms/issues/39538

We all know that one of the problems for newbies when using Joomla is that parameters are stored in multiple places, sometimes they can be overriden. That's why it's good to have help for these parameters. An example might be this problem:

https://github.com/joomla/joomla-cms/issues/39544

There is no information about what function the parameter has.

### Expected result

It is very helpful to have parameter hints

### Actual result

Many parameters lack help.

https://github.com/joomla/joomla-cms/discussions/35594

My opinion (please, it's not a criticism) is still the same: An artificial problem has been created completely unnecessarily. In Joomla 3, the help was in the Tooltip and it didn't bother anyone. Those who wanted to visualize the help got the help, those who didn't, the help didn't bother them at all. And that's even in the case when someone finds specific help completely unnecessary and irrelevant. Since the Tooltip was removed, we started to think about how to solve it differently. The result is an overengineered solution that is not complete at all.

The result - without unnecessary discussions - is the missing help for these parameters. It is not clear what their exact function is.

### System information (as much as possible)

Joomla 4.2.6
PHP 8.1

**Comment by PhocaCz** [comment]
I'm definitely going to do it, as soon as the Tooltips are returned (which in my opinion were senselessly discarded), I'm immediately ready to help with adding hints back to the system.

**Comment by brianteeman** [comment]
I guess your memory is failing you and you have forgotten that Nik added the toggle help button so that you can do this. Just as everyone else who screamed for the tips completely disappeared when the ability to add descriptions was added. No one other than Nik has added any. Guess it really wasnt that important after all.

**Comment by brianteeman** [comment]
This should be closed. There is nothing new here

**Comment by dgrammatiko** [comment]
I will repeat myself here: the inline help button was a mistake, it's inaccessible and with 4.3 introducing (probably) the tour guide (or whatever it name is) it's pretty obvious that was a wrong solution on a problem that the project had a way to fix in a very elegant way. Now Joomla has to support an inaccessible weird interface in parallel with the guided tour...

**Comment by PhocaCz** [comment]
Nik returned the Tooltips? Please read carefully what I write..

**Comment by PhocaCz** [comment]
> I will repeat myself here: the inline help button was a mistake, it's inaccessible and with 4.3 introducing (probably) the tour guide (or whatever it name is) it's pretty obvious that was a wrong solution on a problem that the project had a way to fix in a very elegant way. Now Joomla has to support an inaccessible weird interface in parallel with the guided tour...

I know it goes round and round, but there's just no way to go back and use what Tootlips were just made for?

**Comment by dgrammatiko** [comment]
> I know it goes round and round, but there's just no way to go back and use what Tootlips were just made for?

So let me get this straight:
- Joomla 4 was released with **always visible** descriptions (as long as there was some text)
- You din't like it and Nik introduced an inaccessible interface to hide/show them to fix your problem
- Now you requesting yet another interface (tooltips) for the same thing?
- Joomla is actually planing (it did since my first POC) the guided tour for each page help

My dumb question here: how many interfaces solving the same problem are satisfactory/maintainable/consistent/etc?

**Comment by PhocaCz** [comment]
You keep misinterpreting me. I did not like the removal of Tooltips. Nobody has created anything that solves my problem, because nobody returned back the Tooltips.  And please don't make it sound like Joomla's problems are my problems.

The matter is simple to understand. Yesterday I taught a new Joomla user. I wasn't able to do something as trivial as embedding an image/logo on the site. The problems cycled:

- no display of images ( https://github.com/joomla/joomla-cms/issues/39538 )
- not setting height and width ( https://github.com/joomla/joomla-cms/issues/39543 )
- inability to see help for two similar parameters ( https://github.com/joomla/joomla-cms/issues/39545 )

The training ended in failure. We lost a new Joomla user.

Please don't make it sound like Joomla's problems are my problems.

I, for my self, use Gantry and have no problems with setting logo. So again: **please don't make it sound like Joomla's problems are my problems.**

**Comment by dgrammatiko** [comment]
> please don't make it sound like Joomla's problems are my problems.

@PhocaCz honestly I don't want to fight over this **but** Joomla 4 was released without any problems regarding the description of the input fields, as long as there was some text that was **always** visible. The most accessible and intuitive interface, no javascript no tooltips nothing more required. You (I hope I'm not attributing this falsely to you, if so pardon me) requested some way to hide those descriptions (behind a tooltip as in J3) but what the project accepted (wrongly imho, as I had explained in that issue) the toggle `Inline Help`, which is completely inaccessible and on top of that it doesn't actually fit your expectations. The problem here is getting extremely complicated as the project is planing to release the guided tour (which with fairly little code could act also as something extra to the descriptions), so cumulatively we end up with multiple interfaces for the same problem! Also the tooltips are requested back.
Honestly, if I had to make the call here I would revert the inline help and go back to the always visible descriptions and leave the extra details to the guided tour (with the assumption that forms could provide a simple JSON and not require an extra plugin for the guided tour).
Simplicity is the ultimate sophistication...

**Comment by PhocaCz** [comment]
> You (I hope I'm not attributing this falsely to you, if so pardon me) requested some way to hide those descriptions (behind a tooltip as in J3)

My opinion is clear, completely consistent - not "some way to hide ..." , but returning what worked in Joomla 3 - Tooltips - which are intended for this task

Possible olutions:

a) Tooltips - intended to this task, it appeared from the discussion that they should be OK due to accessibility

b) Help display under the parameter - it's OK, but not for longer help - to the argument that longer help does not belong here, it was clearly answered that people value even this longer help before visiting an external resource and thus interrupting the current activity

c) Toggle Button - partially solves b) but will never be as simple, straightforward and effective as a)

After reading the entire discussion, I simply found a) to be the most watery. Thus, I ask myself the philosophical question why the most suitable solution was removed and replaced with something that constantly causes problems and takes everyone's time. :-(

But back to the topic:

https://github.com/joomla/joomla-cms/issues/39544

The problem here is that we don't know what the parameter actually does, and thanks to a bug we can't test what filling it means. Because entering a value probably won't do anything thanks to a bug

> Honestly, if I had to make the call here I would revert the inline help and go back to the always visible descriptions and leave the extra details to the guided tour

As discussed, "jumping off" at work is very unproductive and distracting. It can completely throw off the workflow. (this opinion is not from my head, it was mentioned in the discussion).

**Comment by dgrammatiko** [comment]
@PhocaCz you do realise that I had coded the tooltip that you're asking for but it was decided to go with the always visible descriptions (and I was fine with that decision). FWIW the preview is here https://joomla-projects.github.io/custom-elements/#/tip and the code is dated back to 2017! adding that code next to any input should be a couple of hours work all and all...

**Comment by wojtekxtx** [comment]
> I don't want to fight over this

@dgrammatiko so simply don't. @PhocaCz suggested what many have suggested before (tooltips) and questions its removal. Which was highly questionable indeed. Above all he is voicing his concerns (which he has every right to do). Nothing more than this.

**Comment by chmst** [comment]
>  I think I know, but want to hear from leader 

no comment. 

It is not a question of who did what. At the end it is always a decision of the whole group of developers, testers, maintainers, release managers.

**Comment by brianteeman** [comment]
sorry I dont respond to trolls

**Comment by wojtekxtx** [comment]
> At the end it is always a decision of the whole group of developers, testers, maintainers, release managers.

@chmst depends on project, but all groups you mentioned wouldnt need to decide upon sth, if there wasnt the need/there was no problem rose by someone..

> sorry I dont respond to trolls

@brianteeman completely your choice that you are entitled to. I respect that. Seriously. Even if you call me s troll (which, btw, I do not agree with). I dont mind being called a troll as I know Im not.

**Comment by coolcat-creations** [comment]
@wojtekxtx what do you mean by to stop spamming? I am not aware of any spam I posted? 😳

**Comment by wojtekxtx** [comment]
@coolcat-creations I sent email and prefer to solve this in private.

**Comment by PhocaCz** [comment]
https://github.com/joomla/joomla-cms/discussions/35594 - This discussion is extensive and a lot of information can be gleaned from it

The intention was probably clear, to clean up the help, delete unnecessary texts and perhaps simplify everything by displaying the text directly below the parameters.

Why am I opening this topic? No, it's not trolling or any other way of creating trouble. I personally don't care who, what, etc.? The decision was simply made. The problem why I opened this topic is simple. It is an unresolved topic, still relevant and problematic.

Just see: https://github.com/joomla/joomla-cms/issues/39545#issuecomment-1368978477

You are facing a possible new Joomler and you are not able to do such a trivial thing as insert an image as a logo in the default template and when you look for help on the parameters, it is nowhere to be found.

![img](https://i.imgur.com/iObp7M2.png)
(As a person with cca15 years of experience with Joomla, I know what the first parameter means, I don't know what the others mean exactly. I personally look into the code, but I don't know for a standard user).

Three possible solutions and their pros and cons listed here:
https://github.com/joomla/joomla-cms/issues/39545#issuecomment-1369062288

a) Tooltips - a feature that was created for these purposes
b) Help displayed under the parameter - not OK for longer text
c) Toggle button - technically complicated and more complex

This problem is still open and unsolved, the summary of a) to c) is not a criticism and does not ask who, what and why, it is just a summary of possible solutions.

This post was created as an issue because it is an issue. If no one on the core team cares, let this post die, after a while someone will close it saying it's an old post.

BTW Technically, Tooltips are accessible in Joomla 4, they work on hover or click events and take the value from an XML file by default.

![img](https://i.imgur.com/lEiouOz.png)
(or as written here: https://github.com/joomla/joomla-cms/issues/39545#issuecomment-1369083385 )

**Comment by brianteeman** [comment]
for what feels like the hundredth time of saying it. If you feel that something needs additional information then add it. No one will object to that.

**Comment by PhocaCz** [comment]
for what feels like the hundredth time of saying it. It is not so easy as it seems.

Someone sometimes decided that the given information was not needed.
![img](https://i.imgur.com/im8lhh9.png)

I'm not the one to decide, and I'm not going to add parameter after parameter and argue with someone if it's relevant information or not. The advantage of Tooltips is that even if you display unnecessary information **for someone**, it doesn't bother anyone. E.g. basically no one clicks on the help for the Title parameter, but the fact that even the unnecessary information is there does not bother anyone and that is the advantage of Tooltips.

Yes, if Tooltips are brought back and it is agreed that information that may be worthless to someone can be displayed there, I will definitely join.

**Comment by brianteeman** [comment]
I now have absolutely no idea what you are talking about and what you want to happen - dioes anyone else understand?

**Comment by wojtekxtx** [comment]
@brianteeman yes, I understamd what @PhocaCz talks about.

**Comment by chmst** [comment]
@brianteeman 
- Descriptions from J3 should be restored when they were useful (from the point of view of a  newbie)
- They should be displayed as tooltips (as we have on status buttons, for example).

I myself sometimes miss descriptions. Then I use to read the code or  the tooltip in a J3 installation but this is not user friendly.

**Comment by dgrammatiko** [comment]
@chmst @PhocaCz @wojtekxtx great, so who's doing the PR that reverts the inline help toggle button and adds the tooltip needed code?

**Comment by chmst** [comment]
I wrote a summary of the discussion, to make clear what is required) nothing else.

And said that I am missing some descriptions (where we can easily make small PRs if needed).

**Comment by brianteeman** [comment]
I would be against restoring anything as a tooltip that is not done in such a way as to indicate one is available. That would be an accessibility fail - there must be an indicator.

Otherwise my view has not changed in 4 years. If you want to add some more text then please submit a pr

**Comment by coolcat-creations** [comment]
But I wrote an indicator in J3 ?

**Comment by brianteeman** [comment]
> But I wrote an indicator in J3 ? Can't this code be used?

of course

**Comment by dgrammatiko** [comment]
> But I wrote an indicator in J3

https://github.com/joomla/joomla-cms/pull/18452 is @coolcat-creations 's PR for anyone wondering

**Comment by coolcat-creations** [comment]
> > But I wrote an indicator in J3
> 
> 
> 
> https://github.com/joomla/joomla-cms/pull/18452 is @coolcat-creations 's PR for anyone wondering

Thank you currently not in the office so it complicated to 
search and reference it :-)

**Comment by wojtekxtx** [comment]
> who's doing the PR

@dgrammatiko: not me. At least not now. Till early next week Im super busy.........

**Comment by dgrammatiko** [comment]
> so what?

You're B/C breaks are not a problem anymore 😕

> @dgrammatiko: not me. At least not now. Till early next week Im super busy.........

Right, https://github.com/joomla/joomla-cms/pulls/wojtekxtx

**Comment by brianteeman** [comment]
>>Otherwise my view has not changed in 4 years. If you want to add some more text then please submit a pr

> I really don't want to lecture and advise, but it is not appropriate to make a political decision first and then possibly do PR?

For the avoidance of any confusion my view has always been that where necessesary there should be "descriptions".

Not sure what you mean about politics

> c) Toggle button - technically complicated and more complex

Could you explain what is technically complicated and more complex about this option. It already exists you just need to add the text as has already It just needs the text

**Comment by PhocaCz** [comment]
>Could you explain what is technically complicated and more complex about this option. It already exists you just need to add the text as has already It just needs the text

https://github.com/joomla/joomla-cms/issues/39545#issuecomment-1369037453

I don't know if I misunderstood this post wrongly, but:

> ... but what the project accepted (wrongly imho, as I had explained in that issue) the toggle Inline Help, which is completely inaccessible ...

Plus other things discussed long ago. Even if you have a toggle button, you still won't get rid of the problem of long help text, problems with a large number of parameters and scrolling to the toggle button, etc. Simply, a toggle button will never be as simple a solution as Tooltips.

c) is not a solution, it is a helper to solve some problems b). Thus, the simple solution a) became a more complicated solution b), which additionally needs c) - technically complicated and more complex.

**Comment by PhocaCz** [comment]
>Not sure what you mean about politics

Making a PR is a technical matter, deciding whether such a PR will be included in the project is a political decision. The political decision should be made before the creation of technical PR. Because otherwise it can turn out like many PRs (including the dark theme of the admin template PR)

**Comment by dgrammatiko** [comment]
> Because otherwise it can turn out like many PRs (including the dark theme of the admin template PR)

Jan, you've mentioned that PR (I'm guessing you're referring to Nik's one here) twice but probably it's the wrong analogy. That PR tried to solve with PHP a problem that is in the Browser territory and in the way created unneeded complexity both for the templates and the editors and on top of that essentially blocking any new Bootstrap updates. Like BS2.1.3 and J3 but actually even worse. Honestly I feel like an idiot right now stepping up and helping (as I did when **you** asked me about Bootstrap 5.0, you do remember that, right?) because I could have let the project go with that solution and laugh from the sidelines seeing Joomla struggling the coming months. Maybe I should close both those PRs https://github.com/joomla/joomla-cms/pull/39379 https://github.com/joomla/joomla-cms/pull/39400 and go spend my time on something else...

**Comment by PhocaCz** [comment]
@dgrammatiko 

Please don't look for any complications here. My opinion (maybe a few people's too) is that simply to show the help the easiest solution is Tooltip.

If I misunderstood or misquoted something somewhere, then I'm sorry.

In no way do I want to criticize anyone's work or anyone's decisions.

I pointed out in this thread that some parameters are missing help, and also that it may be that it has moved away from Tooltips unnecessarily and the current solution is not that good. But that's just my opinion. If someone from the core team reads it and thinks about it, OK. If not, it just will be.

When I talk to people about PR issues, most of the time we agree on why there isn't a brainstorm first, then a political decision, and then the technical execution of the PR.

Personally, I was saddened by PR regarding the dark theme of the admin template. If there had been brainstorming before the technical PR, the PR would certainly not have turned out as badly as it did.

**Comment by PhocaCz** [comment]
Yes, that's why I was just commenting that **maybe** it's better to know if there is any interest in it before doing a PR.

**Comment by wojtekxtx** [comment]
@dgrammatiko nowhere did I say Im busy with Joomla. So no idea how you came to this conclusion.

**Comment by chmst** [comment]
We could make a little change right now: Switch the inline help to visible per default, so that newbies see the information at once and experienced users can switch off.

**Comment by wojtekxtx** [comment]
@chmst sounds like a plan 👍
